Dragen fusions

skanwal opened the pull request:
- updated params
- added ability to read Dragen fusion output
- added Dragen fusion output filtering
- added Dragen fusion output prioritisation

JMarzec commented on rmd_files/RNAseq_report.Rmd:
```diff
@@ -4107,8 +4247,8 @@ if ( runFusionChunk ) {
   fusions.table <- fusions.table[ , names(fusions.table) %!in% "FGID" ]
-  fusions.table <- fusions.table[ , c("geneA", "geneB", "split_readsA", "split_readsB", "discordant_mates", "geneA_dna_support", "geneB_dna_support", "reported_fusion", "fusions_cancer", "reported_fusion_geneA", "reported_fusion_geneB", "confidence", "breakpointA", "breakpointB", "siteA", "siteB", "type", "circos")]
-  names(fusions.table) <- c("Gene A", "Gene B", "Split reads (A)", "Split reads (B)", "Pair reads", "DNA support (A)", "DNA support (B)", "Reported fusion", "Cancer gene(s)", "Fusion gene (A)", "Fusion gene (B)", "Confidence (Arriba)", "Breakpoint (A)", "Breakpoint (B)", "Site (A)", "Site (B)", "Type", "Genomic view")
+  fusions.table <- fusions.table[ , c("geneA", "geneB", "Score", "LeftBreakpoint", "RightBreakpoint", "geneA_dna_support", "geneB_dna_support", "reported_fusion", "fusions_cancer", "reported_fusion_geneA", "reported_fusion_geneB", "circos")]
+  names(fusions.table) <- c("Gene A", "Gene B", "Score", "Left breakpoint", "Right breakpoint", "DNA support (A)", "DNA support (B)", "Reported fusion", "Cancer gene(s)", "Fusion gene (A)", "Fusion gene (B)", "Genomic view")
```
I wonder if we can add "if else" to accomodate differences in content from Arriba/Pizzly and Dragen output

skanwal commented:
@JMarzec - sorry about the delay. I have made the suggested changes. Quick summary is:

- added a separate processing block for processing of Dragen fusions
- Added fusion table summary
- added genomic view circos info
- added various checks for input files (Arriba vs Dragen)

I have tested the changes and these work for me. Let me know if you have any issues.

JMarzec commented on rmd_files/RNAseq_report.Rmd:
````diff
@@ -4225,7 +4225,7 @@ Out of the `r if ( runFusionChunk ) { nrow(fusions) } else { c("0") }` fusion ev
-```{r fusions_summary_table, comment = NA, message=FALSE, warning=FALSE}
+```{r fusions_summary_table_Arriba, comment = NA, message=FALSE, warning=FALSE, eval = runArribaChunk}
 ##### Create a nice table output (with dataTable)
 if ( runArribaChunk ) {
````
I think we should use "runPizzlyChunk" rather than "runArribaChunk" as fusions results do not depend solely on Arriba output, some results can derive from Pizzly and so the chunk should be run

JMarzec commented on rmd_files/RNAseq_report.Rmd:
````diff
@@ -4225,7 +4225,7 @@ Out of the `r if ( runFusionChunk ) { nrow(fusions) } else { c("0") }` fusion ev
 **`r if ( !runFusionChunk ) { c("No fusion genes were detected!") }`**
 
-```{r fusions_summary_table, comment = NA, message=FALSE, warning=FALSE}
+```{r fusions_summary_table_Arriba, comment = NA, message=FALSE, warning=FALSE, eval = runArribaChunk}
````
I think we should use "runPizzlyChunk" rather than "runArribaChunk" as fusions results do not depend solely on Arriba output, some results can derive from Pizzly and so the chunk should be run

JMarzec commented on rmd_files/RNAseq_report.Rmd:
````diff
@@ -4473,7 +4473,7 @@ if ( nrow(fusion_annot_top) > 0 ) {
 
 ```{r genomic_view_circos_table_fusions, comment = NA, message=FALSE, warning=FALSE, eval = runFusionChunk}
-if ( nrow(fusion_annot_top) > 0 ) {
+if ( nrow(fusion_annot_top) > 0 && runArribaChunk ) {
````
As before, possible we should set this one to "runPizzlyChunk" rather than "runArribaChunk"

JMarzec reviewed (commented):
Dragen fusions are added
